sdk/core/azure-core-amqp/src/models/amqp_error.cpp

```diff
@@ -12,11 +12,25 @@
 
 #include <iostream>
 
+#if defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6))
+#pragma GCC diagnostic push
+#elif defined(__clang__) // !__clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdocumentation-unknown-command"
+#endif // _MSC_VER
+
+/// @cond HIDDEN
```

Comment:
These aren't doxygen warnings. They're clang warnings.

And doxygen looks at all files - headers and implementations. That's the way it's worked since it was first brought online, and I'm loathe to change that significant a behavior.

There is something about the template specialization that doxygen is upset about, I believe it's because the type specialization is in the global namespace, but I'm not sure. So I work around it by asking doxygen to skip over the types.

And the failure is happening on the doxygen parse - doxygen compiles the C++ code and then uses the syntax tree generated to find all the documentation. That's *before* it gets to filtering out types.